plugins/nf-cws/src/main/nextflow/cws/wow/filesystem/WOWFileSystemProvider.groovy

```diff
@@ -0,0 +1,235 @@
+        }
+
+        throw new UnsupportedOperationException("Directory stream not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void createDirectory(Path path, FileAttribute<?>... fileAttributes) throws IOException {
+        throw new UnsupportedOperationException("Create directory not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void delete(Path path) throws IOException {
+        throw new UnsupportedOperationException("Delete not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void copy(Path path, Path path1, CopyOption... copyOptions) throws IOException {
+        throw new UnsupportedOperationException("Copy not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void move(Path path, Path path1, CopyOption... copyOptions) throws IOException {
+        throw new UnsupportedOperationException("Move not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    boolean isSameFile(Path path1, Path path2) throws IOException {
+        return path1 == path2
+    }
+
+    @Override
+    boolean isHidden(Path path) throws IOException {
+        return path.getFileName().startsWith(".")
+    }
+
+    @Override
+    FileStore getFileStore(Path path) throws IOException {
+        throw new UnsupportedOperationException("File store not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void checkAccess(Path path, AccessMode... accessModes) throws IOException {
+        // all access is allowed
+    }
+
+    @Override
+    <V extends FileAttributeView> V getFileAttributeView(Path path, Class<V> aClass, LinkOption... linkOptions) {
+        throw new UnsupportedOperationException("File attribute view not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    <A extends BasicFileAttributes> A readAttributes(Path path, Class<A> aClass, LinkOption... linkOptions) throws IOException {
+        if (path instanceof LocalPath) {
+            return path.getAttributes() as A
+        } else {
+            return Files.readAttributes(path, BasicFileAttributes.class, linkOptions) as A
+        }
+    }
+
+    @Override
+    Map<String, Object> readAttributes(Path path, String s, LinkOption... linkOptions) throws IOException {
+        throw new UnsupportedOperationException("Read attributes not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    void setAttribute(Path path, String s, Object o, LinkOption... linkOptions) throws IOException {
+        throw new UnsupportedOperationException("Set attribute not supported by ${getScheme().toUpperCase()} file system provider")
+    }
+
+    @Override
+    boolean canUpload(Path source, Path target) {
+        return false

```

Comment:
remove the return to be consistent:
Methods that return arguments in the first line only have the return value without a return statement.